freqtrade/freqtradebot.py

```diff
@@ -846,24 +848,30 @@ def execute_sell(self, trade: Trade, limit: float, sell_reason: SellType) -> Non
         trade.open_order_id = order_id
         trade.close_rate_requested = limit
         trade.sell_reason = sell_reason.value
+        Trade.session.flush()
+        self.notify_sell(trade)
 
-        profit_trade = trade.calc_profit(rate=limit)
+    def notify_sell(self, trade: Trade):
+        """
+        Sends rpc notification when a sell occured.
+        """
+        profit_trade = trade.calc_profit(rate=trade.close_rate_requested)
         current_rate = self.exchange.get_ticker(trade.pair)['bid']
-        profit_percent = trade.calc_profit_percent(limit)
+        profit_percent = trade.calc_profit_percent(trade.close_rate_requested)
```

Comment:
what prevents you to fetch the real value (average) instead of requested value?

Reply:
because at the time the notification is sent (for regular sell orders, anyway) - the order is not yet completed, only submitted to the exchange.
I know the average after it's fulfilled - but when the notification is sent.

I also cannot use `close_rate` - since that is not known at this point - only once the order is fullfilled.
I'll however change this to use `trade.close_rate` if it's set (for stoploss orders) - and requested-rate otherwise.